src/GithubExtractor.ts

```diff
@@ -0,0 +1,198 @@
+import {Logger} from './Logger'
+import {GithubClientFactory} from "./GithubClientFactory";
+import {Metrics} from "./Metrics";
+import {GithubConfigs} from "./Config";
+import {Octokit} from "@octokit/rest";
+
+export class GithubExtractor {
+    client: Octokit;
+    logger: Logger;
+    metrics: Metrics;
+    config: GithubConfigs;
+
+    constructor(githubClient: Octokit, logger: Logger, metrics: Metrics, config: GithubConfigs) {
+        this.client = githubClient;
+        this.logger = logger;
+        this.metrics = metrics;
+        this.config = config;
+    }
+
+    processRepoPulls(repository): Promise<any> {
```

Comment:
Si ça retourne rien, c'est mieux `void` que `any`.

```suggestion
    processRepoPulls(repository): Promise<void> {
```